cmd/generator/main.go

```diff
@@ -1,11 +1,40 @@
+// +build generate
+
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
 package main
 
 import (
+	"fmt"
+	"os"
+	"path/filepath"
+
 	"github.com/crossplane-contrib/terrajet/pkg/pipeline"
 
 	"github.com/crossplane-contrib/provider-tf-template/config"
 )
 
 func main() {
-	pipeline.Run(config.GetProvider())
+	if len(os.Args) < 2 || os.Args[1] == "" {
```

Comment:
Could we just default to `CWD` if not provided instead of panic?

Reply:
Yes, if it was a flag, but most CLIs I've used usually accept the directory to work on as argument and treat it as required, like `docker build` for example.

Reply:
I would prefer to have a default (`.`) for this and convert it to a flag if required. But fine to keep it as it is if you prefer so.